macros/upload_artifacts_v2.sql

```diff
@@ -63,13 +97,51 @@
 
         )
 
-        {{ flatten_results("raw_data") }};
+        {{ flatten_results("raw_data") }}
+
+    ) as new_data
+    on old_data.artifact_run_id = new_data.artifact_run_id and old_data.artifact_generated_at = new_data.artifact_generated_at
```

Comment:
```suggestion
    on old_data.artifact_run_id = new_data.artifact_run_id and old_data.node_id = new_data.node_id
```

We have a composite key of artifact_run_id and node_id here as it's flattened right?

Reply:
In line with #103 and #94 - we should use `command_invocation_id` (+/- `node_id`) for these merges.